ttnn/cpp/ttnn/operations/data_movement/transpose/device/transpose_program_factory.cpp

```diff
@@ -1848,11 +1849,22 @@ operation::ProgramWithCallbacks transpose_wh_multi_core_sharded(const Tensor& a,
         total_cores,
         tt::tt_metal::ComputeConfig{.fp32_dest_acc_en = fp32_dest_acc_en, .compile_args = compute_compile_time_args});
 
-    uint32_t Wt = shard_spec.shape[1] / TILE_WIDTH;
-    uint32_t Ht = a.get_legacy_shape()[-2] / TILE_HEIGHT;
-    uint32_t HtWt = Ht * Wt;
-    uint32_t N = shard_spec.shape[0] / a.get_legacy_shape()[-2];
-    uint32_t NHtWt = N * HtWt;
+    auto padded_shape = a.get_padded_shape();
+    auto shard_shape = shard_spec.shape;
+
+    uint32_t H = padded_shape[2], W = padded_shape[3];
+    uint32_t Hs = shard_shape[0], Ws = shard_shape[1];
+
+    uint32_t Hts = Hs / tile.tile_shape[0];
+    uint32_t Wts = Ws / tile.tile_shape[1];
```

Comment:
great! would be good to have a test with a custom tile size now

Reply:
I think the transpose_wh_tile LLK only works for full tiles but we'll have to check